File: test/warnings/sql_without_rowid_tables_warning/cpp/SqlWithoutRowIdTablesWarningTest.cpp
#include <string>
#include <vector>

#include "gtest/gtest.h"

#include "test_utils/ZserioErrorOutput.h"

class SqlWithoutRowIdTablesWarningTest : public ::testing::Test
{
protected:
    SqlWithoutRowIdTablesWarningTest()
    :   zserioWarnings("warnings/sql_without_rowid_tables_warning")
    {}

    const test_utils::ZserioErrorOutput zserioWarnings;
};

TEST_F(SqlWithoutRowIdTablesWarningTest, integerPrimaryKey)
{
    const std::string warning = "integer_primary_key_warning.zs:3:11: "
            "Single integer primary key in without rowid table 'WithoutRowIdTable' "
            "brings performance drop.";
    ASSERT_TRUE(zserioWarnings.isPresent(warning));
}

File: test/warnings/sql_without_rowid_tables_warning/python/SqlWithoutRowIdTablesWarningTest.py
import unittest

from testutils import getZserioApi, assertWarningsPresent


class SqlWithoutRowIdTablesWarningTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.warnings = {}
        getZserioApi(
            __file__, "sql_without_rowid_tables_warning.zs", expectedWarnings=1, errorOutputDict=cls.warnings
        )

    def testIntegerPrimaryKey(self):
        assertWarningsPresent(
            self,
            "sql_without_rowid_tables_warning.zs",
            [
                "integer_primary_key_warning.zs:3:11: "
                "Single integer primary key in without rowid table 'WithoutRowIdTable' "
                "brings performance drop."
            ],
        )
